src/v/pandaproxy/schema_registry/sharded_store.cc

```diff
@@ -178,19 +158,55 @@ sharded_store::project_ids(subject_schema schema) {
             s_id = schema.id;
             vlog(plog.debug, "project_ids: using supplied ID {}", s_id.value());
         }
-    } else if (!s_id) {
+    } else if (s_id) {
+        vlog(plog.debug, "project_ids: existing ID {}", s_id.value());
+    }
+
+    // Determine if the subject already has a version that references this
+    // schema, deleted versions are seen.
+    const auto& sub = schema.schema.sub();
+    const auto versions = co_await _store.invoke_on(
+      shard_for(sub),
+      _smp_opts,
+      [sub](auto& s) -> std::vector<subject_version_entry> {
+          auto res = s.get_version_ids(sub, include_deleted::no);
```

Comment:
The comment says "deleted versions are seen" but here we do not include deleted versions. Is that a contradiction?

Reply:
Excellent spot (I noticed it during this PR, too).

There appears to be some conflation between deleted subjects and deleted versions. Here we're excluding deleted subjects, the vector returned is of a type that has a member `deleted` (version), which is ignored (and thus included).

So, the deleted subject is excluded and the deleted version is included.